src/main/java/com/entropyteam/entropay/common/BaseService.java

```diff
@@ -111,8 +109,19 @@ private Collection<Predicate> buildEntityPredicates(Root<Entity> root, Filter fi
         if (MapUtils.isEmpty(filter.getGetByFieldsFilter())) {
             return CollectionUtils.emptyCollection();
         }
-        return filter.getGetByFieldsFilter().entrySet().stream()
-                .map(f -> cb.equal(root.get(f.getKey()), f.getValue())).collect(Collectors.toSet());
+        Collection<Predicate> predicates = filter.getGetByFieldsFilter().entrySet().stream().filter( f -> f.getKey() != "q")
+                        .map(f -> cb.equal(root.get(f.getKey()), f.getValue())).collect(Collectors.toSet());
+
+
+        if(filter.getGetByFieldsFilter().containsKey("q" )){
+            String searchInput = filter.getGetByFieldsFilter().get("q").toLowerCase();
```

Comment:
move the wildcards adding to this line: "%" + filter.getGetByFieldsFilter().get("q").toLowerCase() + "%"